startups/lib/main_9.dart

```diff
@@ -0,0 +1,220 @@
+                  destinations: [
+                    NavigationRailDestination(
+                      icon: Icon(Icons.home),
+                      label: Text('Home'),
+                    ),
+                    NavigationRailDestination(
+                      icon: Icon(Icons.favorite),
+                      label: Text('Favorites'),
+                    ),
+                  ],
+                  selectedIndex: selectedIndex,
+                  onDestinationSelected: (int value) => setState(() {
+                    selectedIndex = value;
+                  }),
+                ),
+              ),
+              Expanded(
+                child: Container(
+                  color: Theme.of(context).colorScheme.secondaryContainer,
+                  child: page,
+                ),
+              ),
+            ],
+          ),
+        );
+      },
+    );
+  }
+}
+
+class GeneratorPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    var appState = context.watch<MyAppState>();
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Expanded(
+          flex: 2,
+          child: ListView(
+            reverse: true,
+            children: [
+              for (var historicalPair in appState.history)
+                Center(
+                  child: Text(historicalPair.asLowerCase),
+                ),
+            ],
+          ),
+        ),
+        SizedBox(height: 16),
+        BigCard(appState.current),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.end,
+          children: [
+            ElevatedButton.icon(
+              icon: Icon(appState.favorites.contains(appState.current)
+                  ? Icons.favorite
+                  : Icons.favorite_border),
+              label: Text('Like'),
+              onPressed: () {
+                appState.toggleCurrentFavorite();
+              },
+            ),
+            SizedBox(width: 8),
+            ElevatedButton.icon(
+              icon: Icon(Icons.refresh),
+              label: Text('Next'),
+              onPressed: () {
+                appState.generateNext();
+              },
+            ),
+            SizedBox(width: 28),
+          ],
+        ),
+        Spacer(),
+      ],
+    );
+  }
+}
+
+class BigCard extends StatelessWidget {
+  final WordPair pair;
+
+  BigCard(this.pair);
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+    var style = theme.textTheme.displayMedium!.copyWith(
+      color: theme.colorScheme.onPrimary,
+    );
+
+    return Card(
+      margin: EdgeInsets.all(24),
+      color: theme.colorScheme.primary,
+      child: Padding(
+        padding: EdgeInsets.all(16),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Text(pair.first,
+                style: style.copyWith(fontWeight: FontWeight.w100)),
```

Comment:
Unsure whether to use `copyWith` or `apply` here.

Reply:
What are the pros and cons of `copyWith` vs `apply`? 

I don't know enough to compare the two. I usually just use `copyWith` because that's what I've seen in use everywhere.

Reply:
I think they're functionally equivalent. `apply` is more about _adjusting_ (e.g. `fontWeightDelta: -2`) and `copyWith` is more about _replacing_ (e.g. `fontWeight: FontWeight.w100`). In the cases here in this app, I think adjusting is more appropriate. I have to say, though, that I also (irrationally!) dislike doing `copyWith` because it _feels_ less performant. Making a copy! (As a matter of fact, `apply` does the exact same thing.)

Your note that you see `copyWith` everywhere makes me think twice about this, though.